bfdd/bfd_packet.c

```diff
@@ -214,52 +214,72 @@ static int ptm_bfd_process_echo_pkt(struct bfd_vrf_global *bvrf, int s)
 
 void ptm_bfd_snd(struct bfd_session *bfd, int fbit)
 {
-	struct bfd_pkt cp;
+	struct bfd_pkt cp = {0};
+	struct bfd_pkt *pst_bfd_pkt = NULL;
 
-	/* Set fields according to section 6.5.7 */
-	cp.diag = bfd->local_diag;
-	BFD_SETVER(cp.diag, BFD_VERSION);
-	cp.flags = 0;
-	BFD_SETSTATE(cp.flags, bfd->ses_state);
+	if ((true == bfd->bfd_tx_pkt_stored) &&
```

Comment:
Suggestion: use a `goto` to jump to packet send instead of creating a whole new level of code indentation and a lot of modified lines.

Example:

```c
if (have_buffered_packet) {
  packet = bs->packet;
  goto send_packet;
} else
  packet = local_buffer;

/* leave all old code unmodified here. */

_ptm_bfd_send(..., packet, ...);
```